src/main/java/com/johan/blignaut/entelect/vitalityincubator/task2/exception/CustomApiExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.johan.blignaut.entelect.vitalityincubator.task2.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import com.johan.blignaut.entelect.vitalityincubator.restclient.ApiException;
+
+@ControllerAdvice
```

Comment:
This does work, 100%, I do like this but for API exception, I'd rather update it to a runtime exception and add the http status annotations to it. So when you throw the runtime, you don't have to handle it and spring automatically sets the status etc, you can have a message field in the runtime excpetion and spring will response with that in the payload response.

Reply:
Alright cool, I'll give this a try, thanks Vince.